applications/serial_lte_modem/src/slm_uart_handler.c

```diff
@@ -450,22 +367,17 @@ int slm_uart_handler_init(slm_uart_rx_callback_t callback_t)
 
 	k_sem_give(&tx_done_sem);
 
-	err = slm_uart_tx_write(SLM_SYNC_STR, sizeof(SLM_SYNC_STR)-1, true, false);
-	if (err) {
-		return err;
-	}
+	/* Flush possibly pending data in case SLM was idle. */
+	tx_start();
 
 	return 0;
 }
 
-void slm_uart_handler_uninit(void)
+int slm_uart_handler_enable(void)
 {
-	int err;
-
-	/* Power off UART module */
-	slm_uart_rx_disable();
-	err = pm_device_action_run(uart_dev, PM_DEVICE_ACTION_SUSPEND);
-	if (err) {
-		LOG_WRN("Can't suspend UART: %d", err);
-	}
+	return slm_at_set_backend((struct slm_at_backend) {
+		.start = slm_uart_handler_init,
+		.send = slm_uart_tx_write,
+		.stop = slm_uart_rx_disable
```

Comment:
How about adding a call to `uart_tx_abort()` after the 100ms wait? Would that make sense?

Reply:
It would likely return -EFAULT, which we would need to swallow. Also the uart_callback would just start another TX-operation. So we would need to exit if aborted. And perhaps only if we are closing as it makes sense to retry with other aborts, so this would be bit messy.

Let's add this if we run into problems.